### PR TITLE
tenant manager: tenants statuses metrics

### PR DESCRIPTION
**Pull request by famartinrh**
adds various metrics to tenant-manager for tracking the total number of tenants in each status and the time tenants spend in the TO_BE_DELETED status

### CODE REVIEW

**Review by EricWittmann — APPROVED** [review]
lgtm

File: multitenancy/tenant-manager-api/src/main/java/io/apicurio/multitenant/metrics/UsageMetrics.java [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+import io.micrometer.core.instrument.Tags;
+import io.micrometer.core.instrument.Timer;
+import io.quarkus.runtime.Startup;
+
+/**
+ * @author Fabian Martinez
+ */
+@ApplicationScoped
+@Startup
+public class UsageMetrics {
```
Comment (jsenko) [inline multitenancy/tenant-manager-api/src/main/java/io/apicurio/multitenant/metrics/UsageMetrics.java:47]: Nice code! :)

**Review by jsenko — APPROVED** [review]

File: multitenancy/tenant-manager-api/src/main/java/io/apicurio/multitenant/api/services/TenantStatusService.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+import io.apicurio.multitenant.storage.dto.RegistryTenantDto;
+import io.apicurio.multitenant.utils.Pair;
+
+/**
+ * @author Fabian Martinez
+ */
+@ApplicationScoped
+public class TenantStatusService {
+
+    private List<Pair<TenantStatusValue, TenantStatusValue>> allowedTransitions;
```
[the remark this answers is not in the document]
Comment (famartinrh) [reply multitenancy/tenant-manager-api/src/main/java/io/apicurio/multitenant/api/services/TenantStatusService.java:38]: it's not the case, but my thinking was one source state can lead to multiple destination states

File: multitenancy/tenant-manager-api/src/main/java/io/apicurio/multitenant/metrics/UsageMetrics.java (hunk 1 above)
Comment (famartinrh) [reply multitenancy/tenant-manager-api/src/main/java/io/apicurio/multitenant/metrics/UsageMetrics.java:47]: :) sounds familiar right?

**Pull request merged**